products/batch_exports/backend/temporal/record_batch_model.py

```diff
@@ -0,0 +1,212 @@
+import abc
+import datetime as dt
+import typing
+
+from posthog.batch_exports.service import (
+    BatchExportModel,
+    BatchExportSchema,
+)
+from posthog.hogql.context import HogQLContext
+from posthog.hogql.database.database import create_hogql_database
+from posthog.hogql.hogql import ast
+from posthog.hogql.printer import prepare_ast_for_printing, print_prepared_ast
+from posthog.models import Team
+from posthog.sync import database_sync_to_async
+from products.batch_exports.backend.temporal import sql
+
+Query = str
+QueryParameters = dict[str, typing.Any]
+BatchExportDateRange = tuple[dt.datetime | None, dt.datetime]
+
+
+class RecordBatchModel(abc.ABC):
+    """Base class for models that can be produced as record batches."""
+
+    def __init__(self, team_id: int):
+        self.team_id = team_id
+
+    async def get_hogql_context(self, team_id: int) -> HogQLContext:
+        """Return a HogQLContext to generate a ClickHouse query."""
+        team = await Team.objects.aget(id=team_id)
```

Comment:
I don't see the point of this being a method: None of the class state is being modified or accessed. Couldn't we either make this a function or use `self.team_id`? What do you think?

Reply:
yeah I agree (though I just copied this code over from the old module). Maybe I'll just use `self.team_id` then